### PR TITLE
[TACHYON-969] Sets up the tiered storage hierarchy only once for each test class if possible

### PR DESCRIPTION
**Pull request by fanshiqing**
https://tachyon.atlassian.net/browse/TACHYON-969

Currently, we simply set up the tiered storage hierarchy for each test unit method, this is unnecessary in most cases.

Sets up the tiered storage hierarchy only once for each test class if possible.

### CODE REVIEW

**Comment by yupeng9** [comment]
LGTM. Thanks for the test improvement!

File: servers/src/test/java/tachyon/worker/block/BlockMetadataManagerTest.java [hunk 1]
```diff
@@ -52,18 +54,21 @@
...
+  public static TemporaryFolder sFolder = new TemporaryFolder();
 
   @Rule
   public ExpectedException mThrown = ExpectedException.none();
 
-  @Before
-  public void before() throws Exception {
-    String baseDir = mFolder.newFolder().getAbsolutePath();
+  @BeforeClass
+  public static void setupTieredStorage() throws Exception {
```
Comment (calvinjia) [inline servers/src/test/java/tachyon/worker/block/BlockMetadataManagerTest.java]: The convention for methods annotated with `@BeforeClass` is to name them `beforeClass`

**Comment by jsimsa** [comment]
What is the benefit of this change? Is it needed to make the tests run faster?

In general, sharing state (e.g. the TempFolder or the tiered storage hierarchy) between tests can lead to unexpected behavior (e.g. if some test does not clean up after itself and a subsequent test expects a clean state). The fact that the order in which individual tests are executed can change over time also means that such errors can be masked until a completely unrelated test is added to the test suite.

**Comment by apc999** [comment]
Agree what @jsimsa says, sometimes it is more important for us to know unit tests cases are isolated without sharing states, so we could reason about failure easier.  Actually it happens to us before that we fight with unit test failures due to unnecessary state sharing among different test cases.

**Comment by apc999** [comment]
I guess that the general rule of thumb is, unless the performance is really a big issue of a given unit tests, make sure it is clean and easier to maintain is more important.

**Comment by fanshiqing** [comment]
@apc999 @jsimsa Thanks for your detailed explanation :+1: ! Actually where I changed to share the tiered hierarchy will not encounter conflicts as expected, what jsimsa has said was an important consideration for me before this PR. 

While I will close this PR to prevent potential risks. Thanks :)

**Pull request closed without merging**